### PR TITLE
test: add comprehensive unit tests for UrlSigner

### PR DESCRIPTION
**Pull request by farion1231**
- Add tests for signed URL parameter generation
- Add tests for signature verification with various scenarios
- Add tests for security aspects (nonce randomness, signature tampering)
- Add tests for edge cases (empty strings, special characters)
- Ensure proper error handling when SECRET_KEY is not configured

## Checklist

- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] I understand that this PR may be closed in case there was no previous discussion or issues. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly.
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

### CODE REVIEW

**Review by crazywoola — DISMISSED** [review]

File: api/tests/unit_tests/core/helper/test_url_signer.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        nonces = set()
+        for _ in range(5):
+            params = UrlSigner.get_signed_url_params(sign_key, prefix)
+            nonces.add(params.nonce)
+
+        # All nonces should be unique
+        assert len(nonces) == 5
+
+    @patch("configs.dify_config.SECRET_KEY", "test-secret-key-12345")
+    def test_should_produce_consistent_signatures(self):
```
[the remark this answers is not in the document]
Comment (farion1231) [reply api/tests/unit_tests/core/helper/test_url_signer.py]: Agree, I've refactored the test to use the public API instead of calling the private `_sign` method directly.

File: api/tests/unit_tests/core/helper/test_url_signer.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    def test_should_generate_complete_signed_url(self):
+        """Test generation of complete signed URL with query parameters"""
+        base_url = "https://example.com/api/test"
+        sign_key = "test-sign-key"
+        prefix = "test-prefix"
+
+        signed_url = UrlSigner.get_signed_url(base_url, sign_key, prefix)
+
+        # Verify URL contains all required parameters
+        assert signed_url.startswith(base_url)
```
Comment (QuantumGhost) [inline api/tests/unit_tests/core/helper/test_url_signer.py]: IMO it's better to parse the signed url and assert the query string contains `timestamp`, `nonce` and `sign`.

**Review by QuantumGhost — DISMISSED** [review]
Other changes LGTM

File: api/tests/unit_tests/core/helper/test_url_signer.py (hunk 2 above)
Comment (farion1231) [reply api/tests/unit_tests/core/helper/test_url_signer.py]: Thanks! I've updated the test to properly parse the URL and validate the query parameters structure.

**Review by QuantumGhost — APPROVED** [review]
LGTM.

Thank you for your contribution.

**Comment by farion1231** [comment]
> LGTM.
> 
> Thank you for your contribution.

Thank you for maintaining this great project!
